File: microservices/payment-service/src/main/java/com/karate/payment_service/infrastructure/client/AuthClient.java
package com.karate.payment_service.infrastructure.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "auth-service")
public interface AuthClient {
    @GetMapping("/internal/users/username/by-id/{username}")
    Long getUserIdByUsername(@PathVariable String username);
}

File: microservices/feedback-service/src/main/kotlin/com/karate/feedback_service/infrastructure/client/AuthClient.kt
package com.karate.feedback_service.infrastructure.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "auth-service")
interface AuthClient {
    @GetMapping("/internal/users/username/by-id/{username}")
    fun getUserIdByUsername(@PathVariable username: String): Long?
}
